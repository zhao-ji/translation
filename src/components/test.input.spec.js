import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Input, { Suggestions } from './input';

describe('<Suggestions />', ()=>{
    describe('render()', () => {
        test('default correctly', () => {
            const props = {
                matchedOption: "hello",
                currentText: "hey",
                onClickSuggestion: () => {},
                cache: {},
            };
            const component = shallow(<Suggestions {...props}/>);
            expect(toJson(component)).toMatchSnapshot();
        });
        test('render suggestions', () => {
            const props = {
                matchedOption: "",
                currentText: "hello",
                onClickSuggestion: () => {},
                cache: {
                    hello: {
                        result: ["hello", "hello world"],
                    },
                },
            };
            const component = shallow(<Suggestions {...props}/>);
            expect(toJson(component)).toMatchSnapshot();
        });
        test('suggestions does not match', () => {
            const props = {
                matchedOption: "hello",
                currentText: "hey",
                onClickSuggestion: () => {},
                cache: {
                    hello: {
                        result: ["hello", "hello world"],
                    },
                },
            };
            const component = shallow(<Suggestions {...props}/>);
            expect(toJson(component)).toMatchSnapshot();
        });
    });
});

describe('<Input />', ()=>{
    describe('render()', () => {
        test('default correctly', () => {
            const props = {
            };
            const component = shallow(<Input {...props}/>);
            expect(toJson(component)).toMatchSnapshot();
        });
    });
});
