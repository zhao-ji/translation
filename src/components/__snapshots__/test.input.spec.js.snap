// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> render() default correctly 1`] = `
<Fragment>
  <Bootstrap(InputGroup)
    id="search-component"
    size="lg"
  >
    <FormControl
      as="input"
      autoComplete="off"
      autoFocus={true}
      defaultValue=""
      id="input-search-bar"
      key=""
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="translate word or sentence for english and mandarin..."
      type="search"
    />
    <InputGroupAppend>
      <Button
        active={false}
        block={true}
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="outline-dark"
      >
        Search
      </Button>
    </InputGroupAppend>
  </Bootstrap(InputGroup)>
  <Suggestions
    matchedOption=""
    onClickSuggestion={[Function]}
  />
</Fragment>
`;

exports[`<Suggestions /> render() default correctly 1`] = `<Fragment />`;

exports[`<Suggestions /> render() render suggestions 1`] = `
<ForwardRef>
  <Bootstrap(ListGroupItem)
    action={true}
    className="option"
    data-value="hello"
    key="0"
    onClick={[Function]}
  >
    hello
    <b />
  </Bootstrap(ListGroupItem)>
  <Bootstrap(ListGroupItem)
    action={true}
    className="option"
    data-value="hello world"
    key="1"
    onClick={[Function]}
  >
    hello
    <b>
       world
    </b>
  </Bootstrap(ListGroupItem)>
</ForwardRef>
`;

exports[`<Suggestions /> render() suggestions does not match 1`] = `<Fragment />`;
